src/client/datascience/data-viewing/types.ts

```diff
@@ -38,29 +40,39 @@ export interface IGetRowsResponse {
     end: number;
 }
 
+export interface IGetSliceRequest {
+    slice: string;
+    originalShape: number[] | undefined;
+}
+
 // Map all messages to specific payloads
 export type IDataViewerMapping = {
     [DataViewerMessages.Started]: never | undefined;
     [DataViewerMessages.UpdateSettings]: string;
-    [DataViewerMessages.InitializeData]: IDataFrameInfo;
-    [DataViewerMessages.GetAllRowsRequest]: never | undefined;
+    [DataViewerMessages.InitializeData]: IDataFrameInfo & { isSliceDataSupported: boolean };
+    [DataViewerMessages.GetAllRowsRequest]: never | undefined | string;
     [DataViewerMessages.GetAllRowsResponse]: IRowsResponse;
     [DataViewerMessages.GetRowsRequest]: IGetRowsRequest;
     [DataViewerMessages.GetRowsResponse]: IGetRowsResponse;
     [DataViewerMessages.CompletedData]: never | undefined;
+    [DataViewerMessages.GetSliceRequest]: IGetSliceRequest;
 };
 
 export interface IDataFrameInfo {
     columns?: { key: string; type: ColumnType }[];
     indexColumn?: string;
     rowCount?: number;
+    shape?: number[];
+    dataDimensionality?: number;
```

Comment:
This property isn't being used yet, leaving it in here because it'll be useful when building UI (for controlling whether to show slice controls for some data)